src/libs/actions/Task.js

```diff
@@ -233,4 +345,55 @@ function clearOutTaskInfoAndNavigate(reportID) {
     Navigation.navigate(ROUTES.NEW_TASK_DETAILS);
 }
 
-export {createTaskAndNavigate, setTitleValue, setDescriptionValue, setDetailsValue, setAssigneeValue, setShareDestinationValue, clearOutTaskInfo, clearOutTaskInfoAndNavigate};
+/**
+ * Get the assignee data
+ *
+ * @param {Object} details
+ * @returns {Object}
+ */
+function getAssignee(details) {
+    if (!details) {
+        return {
+            icons: [],
+            displayName: '',
+            subtitle: '',
+        };
+    }
+    const source = ReportUtils.getAvatar(lodashGet(details, 'avatar', ''), lodashGet(details, 'login', ''));
+    return {
+        icons: [{source, type: 'avatar', name: details.login}],
+        displayName: details.displayName,
+        subtitle: details.login,
+    };
+}
+
+/**
+ * Get the share destination data
+ * @param {Object} reportID
+ * @param {Object} reports
+ * @param {Object} personalDetails
+ * @returns {Object}
+ * */
+function getShareDestination(reportID, reports, personalDetails) {
+    const report = lodashGet(reports, `report_${reportID}`, {});
+    return {
+        icons: ReportUtils.getIcons(report, personalDetails),
```

Comment:
✋ Coming from https://github.com/Expensify/App/issues/21582

The task can be created inside the money request thread. So we should get the correct avatar by passing the `isPayer` from `ReportUtils.isIOUReport` result.